Sources/Rendering/SceneGraph/ViewNodeFactory/index.js

```diff
@@ -15,20 +15,18 @@ function vtkViewNodeFactory(publicAPI, model) {
     if (dataObject.isDeleted()) {
       return null;
     }
-    // if (Object.keys(model.overrides).indexOf(dataObject.getClassName()) === -1) {
-    //   return null;
-    // }
 
-    const hierarchy = dataObject.getHierarchy();
+    let cpt = 0;
+    let className = dataObject.getClassName(cpt++);
     let isObject = false;
-    let className = '';
-    for (let i = hierarchy.length - 1; i > 0; i--) {
-      if (Object.keys(model.overrides).indexOf(hierarchy[i]) !== -1) {
+    while (className) {
+      if (Object.keys(model.overrides).indexOf(className) !== -1) {
```

Comment:
Try to cache `Object.keys(model.overrides)` instead of recomputing each time inside a loop.

Reply:
Since you have a while, you can put your break statement in that `while()`